org.eclipse.draw2d/src/org/eclipse/draw2d/graph/EdgeList.java

```diff
@@ -14,100 +14,93 @@
 
 /**
  * A list of <code>Edge</code>s.
- * 
+ *
  * @author hudsonr
  * @since 2.1.2
  */
-public class EdgeList extends ArrayList {
+public class EdgeList extends ArrayList<Edge> {
 
 	/**
 	 * Returns the edge for the given index.
-	 * 
+	 *
 	 * @param index the index of the requested edge
 	 * @return the edge at the given index
+	 *
+	 * @deprecated use Use {@link #get(int)} instead.
 	 */
+	@Deprecated(since = "3.13.100", forRemoval = true)
 	public Edge getEdge(int index) {
-		return (Edge) super.get(index);
+		return super.get(index);
 	}
 
 	/**
 	 * For intrenal use only.
-	 * 
+	 *
 	 * @param i and index
 	 * @return a value
 	 */
 	public int getSourceIndex(int i) {
-		return getEdge(i).source.index;
+		return get(i).source.index;
 	}
 
 	/**
 	 * For internal use only.
-	 * 
+	 *
 	 * @param i an index
 	 * @return a value
 	 */
 	public int getTargetIndex(int i) {
-		return getEdge(i).target.index;
+		return get(i).target.index;
 	}
 
 	/**
 	 * For internal use only.
-	 * 
+	 *
 	 * @return the minimum slack for this edge list
 	 */
 	public int getSlack() {
-		int slack = Integer.MAX_VALUE;
-		for (int i = 0; i < this.size(); i++)
-			slack = Math.min(slack, getEdge(i).getSlack());
-		return slack;
+		return stream().mapToInt(Edge::getSlack).min().orElse(Integer.MAX_VALUE);
 	}
 
 	/**
 	 * For internal use only.
-	 * 
+	 *
 	 * @return the total weight of all edges
 	 */
 	public int getWeight() {
-		int w = 0;
-		for (int i = 0; i < this.size(); i++)
-			w += getEdge(i).weight;
-		return w;
+		return stream().mapToInt(e -> e.weight).sum();
 	}
 
 	/**
 	 * For internal use only
-	 * 
+	 *
 	 * @return <code>true</code> if completely flagged
 	 */
 	public boolean isCompletelyFlagged() {
-		for (int i = 0; i < size(); i++) {
-			if (!getEdge(i).flag)
-				return false;
-		}
-		return true;
+		return stream().noneMatch(e -> !e.flag);
```

Comment:
As mentioned earlier, I think an allMatch(e -> e.flag) is more readable.

Reply:
This should be part of a separate pull request. Will fix it there.